content/automate/ManageIQ/Transformation/TransformationHosts/Common.class/__methods__/killvirtv2v.rb

```diff
@@ -1,28 +1,38 @@
 module ManageIQ
   module Automate
     module Transformation
-      module TransformationHost
+      module TransformationHosts
         module Common
           class KillVirtV2V
             def initialize(handle = $evm)
               @handle = handle
+              @task = ManageIQ::Automate::Transformation::Common::Utils.task(@handle)
             end
 
-            def task_virtv2v_state(task, transformation_host)
+            def task_virtv2v_state(transformation_host)
               require 'json'
-              return if task.get_option(:virtv2v_started_on).blank? || task.get_option(:virtv2v_finished_on).present?
-              return if task.get_option(:virtv2v_wrapper).blank?
-              result = Transformation::TransformationHosts::Common::Utils.remote_command(task, transformation_host, "cat '#{task.get_option(:virtv2v_wrapper)['state_file']}'")
+              return if @task.get_option(:virtv2v_started_on).blank? || @task.get_option(:virtv2v_finished_on).present?
+              return if @task.get_option(:virtv2v_wrapper).blank?
+              result = ManageIQ::Automate::Transformation::TransformationHosts::Common::Utils.remote_command(@task, transformation_host, "cat '#{@task.get_option(:virtv2v_wrapper)['state_file']}'")
               return if !result[:success] || result[:stdout].empty?
               JSON.parse(result[:stdout])
             end
 
+            def kill_virtv2v(transformation_host, pid)
+              signal = 'KILL'
+              unless @handle.get_state_var('virtv2v_graceful_kill')
+                signal = 'TERM'
+                @handle.set_state_var('virtv2v_graceful_kill', true)
+                @handle.root['ae_result'] = 'retry'
+                @handle.root['ae_retry_interval'] = '30.seconds'
+              end
+              ManageIQ::Automate::Transformation::TransformationHosts::Common::Utils.remote_command(@task, transformation_host, "kill -s #{signal} #{pid}")
```

Comment:
I was thinking about how to write this without using `unless` since we normally only use that for simple one-line conditions.

My suggestion would be to move the `kill_signal` logic into a separate method.

```ruby
def kill_virtv2v(transformation_host, pid)
  ManageIQ::Automate::Transformation::TransformationHosts::Common::Utils.remote_command(@task, transformation_host, "kill -s #{kill_signal} #{pid}")
end

def kill_signal
  if @handle.get_state_var('virtv2v_graceful_kill')
    'KILL'
  else
    @handle.set_state_var('virtv2v_graceful_kill', true)
    @handle.root['ae_result'] = 'retry'
    @handle.root['ae_retry_interval'] = '30.seconds'
    'TERM'
  end
end
```